addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -4417,22 +4417,17 @@ var SnippetsMenu = Widget.extend({
      * Preview on mobile.
      */
     _onMobilePreviewClick() {
-        this.trigger_up('request_mobile_preview');
-
-        // TODO refactor things to make this more understandable -> on mobile
-        // edition, update the UI. But to do it properly and inside the mutex
-        // this simulates what happens when a snippet option is used.
+        this._execWithLoadingEffect(async () => this.trigger_up('request_mobile_preview'), false);
         this._execWithLoadingEffect(async () => {
-            // TODO needed so that mobile edition is considered before updating
-            // the UI but this is clearly random. The trigger_up above should
-            // properly await for the rerender somehow.
-            await new Promise(resolve => setTimeout(resolve));
-
-            return new Promise(resolve => {
+            // TODO: the trigger_up above should properly await for the
+            // rerender somehow.
+            // SetTimeout needed so that the mobile preview is considered
+            // before updating the UI with snippet_option_update.
+            new Promise(resolve => setTimeout(() => {
```

Comment:
`setTimeout` is probably not the way to go... but here you don't even return the Promise, should not you? :grimacing:

Reply:
- `setTimeout` is what was used before, just not exactly written in the same way. Without it, the overlays are consistently reversed (i.e. the mobile overlays are set on desktop and the desktop overlays on mobile). So in the end, it makes the code wait for everything mobile-related to kick in before the `snippet_option_update`.
- removing the return statement is part of the solution to the bug. Everything I tried keeping the return generated the bug.

Reply:
> removing the return statement is part of the solution to the bug. Everything I tried keeping the return generated the bug.

That may be so but... we cannot rely on that without an explanation. Here your promise just does nothing... What you wrote:
```javascript
new Promise(resolve => setTimeout(() => {
    this.trigger_up('snippet_option_update', {
        onSuccess: () => resolve(),
    });
}));
```
is 100% the same as:
```javascript
setTimeout(() => this.trigger_up('snippet_option_update'));
```

So if this is the right solution... it should be written that way. And if it is the right solution, we need a comment as to why this should not be awaited :wink:

Reply:
I pushed an update on this (with `onSuccess: () => null` though, because it's needed), but

> we cannot rely on that without an explanation

I really don't have an answer to that. :grimacing: